### PR TITLE
Add 'clamp' helper function

### PR DESCRIPTION
**Pull request by 5outh**
Adds 'clamp', which constrains an integral value to a closed interval.

### CODE REVIEW

File: library/Closed/Internal.hs [hunk 1]
```diff
@@ -242,3 +242,10 @@ multiply (Closed x) (Closed y) = Closed $ x * y
 isValidClosed :: (KnownNat n, KnownNat m) => Closed n m -> Bool
 isValidClosed cx@(Closed x) =
   natVal (lowerBound cx) <= x && x <= natVal (upperBound cx)
+
+-- | Clamp an @'Integral'@ in the range constrained by a @'Closed'@ interval
+clamp :: forall n m a. (KnownNat n, KnownNat m, n <= m, Integral a) => a -> Closed n m
+clamp x
+  | fromIntegral x < getClosed (minBound @(Closed n m)) = minBound
+  | fromIntegral x > getClosed (maxBound @(Closed n m)) = maxBound
+  | otherwise = Closed (fromIntegral x)
```
Comment (cdparks) [inline library/Closed/Internal.hs]: Nice! Can you move it up with the other constructors (`closed`, `unsafeClosed`) and add an example/test to `README.md`?

Comment (5outh) [reply library/Closed/Internal.hs]: Done! I was wondering where the tests were. Does hspec-discover handle running these?

Comment (cdparks) [reply library/Closed/Internal.hs]: Nah, we have a symbolic link to `README.md` called `README.lhs` and we treat that as the `Main` module for the test stanza. [See here](https://github.com/freckle/closed/blob/master/package.yaml#L28)

File: library/Closed/Internal.hs [hunk 2]
```diff
@@ -1,40 +1,40 @@
...
+{-# LANGUAGE CPP                  #-}
+{-# LANGUAGE DataKinds            #-}
+{-# LANGUAGE DeriveGeneric        #-}
+{-# LANGUAGE ExplicitForAll       #-}
+{-# LANGUAGE FlexibleContexts     #-}
+{-# LANGUAGE KindSignatures       #-}
+{-# LANGUAGE ScopedTypeVariables  #-}
+{-# LANGUAGE TypeApplications     #-}
+{-# LANGUAGE TypeFamilies         #-}
+{-# LANGUAGE TypeOperators        #-}
```
Comment (cdparks) [inline library/Closed/Internal.hs]: I'm fine with sorting these, but can you remove the extra whitespace here and in the imports?

Comment (5outh) [reply library/Closed/Internal.hs]: I just went ahead and cherry-picked the changes; the hlint/brittany files from megarepo are too bloated for this project I think

**Pull request merged**